program/src/instruction.rs

```diff
@@ -1,184 +1,373 @@
-//! Program instructions
+use borsh::{BorshDeserialize, BorshSerialize};
+use shank::{ShankContext, ShankInstruction};
 
-use solana_program::clock::UnixTimestamp;
-
-/// Instructions supported by the staking program
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub enum StakeInstruction {
-    /// Creates Stake config account which controls staking parameters
-    ///
-    /// 0. `[w]` Stake config account
-    /// 1. `[]` Slash authority, may be a PDA for a governance program
-    /// 2. `[]` Config authority, may be a PDA for a governance program
-    /// 3. `[]` Stake Token Mint
-    /// 4. `[]` Stake Token Vault, to hold all staked tokens.
-    ///   Must be empty and owned by PDA with seeds `['token-owner', stake_config]`
+/// Enum defining all instructions in the Stake program.
+#[rustfmt::skip]
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, ShankContext, ShankInstruction)]
+pub enum Instruction {
+    /// Creates Stake config account which controls staking parameters.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        name = "config_authority",
+        desc = "Config authority"
+    )]
+    #[account(
+        2,
+        name = "slash_authority",
+        desc = "Slash authority"
+    )]
+    #[account(
+        3,
+        name = "mint",
+        desc = "Stake token mint"
+    )]
+    #[account(
+        4,
+        name = "vault_token",
+        desc = "Stake token vault"
+    )]
     InitializeConfig {
-        cooldown_time: UnixTimestamp,
+        cooldown_time_seconds: u64,
         max_deactivation_basis_points: u16,
     },
 
     /// Initializes stake account data for a validator.
-    ///
-    /// NOTE: Anybody can create the stake account for a validator. For new
-    /// accounts, the authority is initialized to the validator vote account's
-    /// withdraw authority.
-    ///
-    /// 0. `[]` Stake config account
-    /// 1. `[w]` Validator stake account
-    ///     * PDA seeds: ['stake', validator, config_account]
-    /// 2. `[]` Validator vote account
-    /// 3. `[]` System program
+    #[account(
+        0,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        writable,
+        name = "stake",
+        desc = "Validator stake account (pda of `['stake', config, validator]`)"
```

Comment:
The last two seeds here are backwards. The original comment said `['stake', validator, config_account]`.

Reply:
Good catch, thanks!